com.hazelcast.collection.impl.queue.QueuePriorityTest.testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain [HZ-1839]

_4.2.z_ (commit e38d1be32a033f9ed5cfa7f8d109e52ec29dc081)

Failed on OracleJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.maintenance-OracleJDK8/180/testReport/com.hazelcast.collection.impl.queue/QueuePriorityTest/testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<2> but was:<1>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.collection.impl.queue.QueuePriorityTest.testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain(QueuePriorityTest.java:128)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
18:14:07,651  INFO |queueConsistency| - [MetricsConfigHelper] Thread-51873 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
Finished Running Test: queueParallel in 1.546 seconds.
18:14:07,652  WARN |queueConsistency| - [CPSubsystem] Thread-51873 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
18:14:07,654  INFO |queueConsistency| - [Diagnostics] Thread-51873 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
18:14:07,654  INFO |queueConsistency| - [LifecycleService] Thread-51873 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] [127.0.0.1]:5702 is STARTING
18:14:07,654  INFO |queueConsistency| - [MockServer] Thread-51873 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5701, alive=true}
18:14:07,654  INFO |queueConsistency| - [MockServer] hz.compassionate_hofstadter.generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5702, alive=true}
18:14:07,655  INFO |queueConsistency| - [ClusterService] hz.compassionate_hofstadter.generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 1d15f5c0-a36e-42bd-9b5a-699a8aec1c24 this
	Member [127.0.0.1]:5702 - f72654a1-7907-4319-8ff2-863d6540f642
]

18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [LifecycleService] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] [127.0.0.1]:5703 is SHUTTING_DOWN
18:14:07,663  WARN |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] Terminating forcefully...
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] Shutting down connection manager...
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MockServer] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5703, alive=false}
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MockServer] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5702, alive=false}
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MockServer] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5703, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5703, alive=false}
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MockServer] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5703, remoteEndpoint=[127.0.0.1]:5701, alive=false}
18:14:07,663  WARN |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MembershipManager] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Member [127.0.0.1]:5703 - 70085a55-69e7-4f11-acee-694cfd5a6f01 is suspected to be dead for reason: Connection manager is stopped on Member [127.0.0.1]:5703 - 70085a55-69e7-4f11-acee-694cfd5a6f01 this
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MembershipManager] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Removing Member [127.0.0.1]:5703 - 70085a55-69e7-4f11-acee-694cfd5a6f01
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [ClusterService] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - bee6fea1-44c0-43fb-ab92-a8f9657c732e this
	Member [127.0.0.1]:5702 - 04b9ca9c-a44b-4878-b63e-0cdb505b0b52
]

18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] Shutting down node engine...
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [TransactionManagerService] hz.sharp_hofstadter.cached.thread-3 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 70085a55-69e7-4f11-acee-694cfd5a6f01
18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [ClusterService] hz.nervous_hofstadter.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - bee6fea1-44c0-43fb-ab92-a8f9657c732e
	Member [127.0.0.1]:5702 - 04b9ca9c-a44b-4878-b63e-0cdb505b0b52 this
]

18:14:07,663  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [TransactionManagerService] hz.nervous_hofstadter.cached.thread-6 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: 70085a55-69e7-4f11-acee-694cfd5a6f01
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [NodeExtension] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] Destroying node NodeExtension.
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] Hazelcast Shutdown is completed in 1 ms.
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [LifecycleService] Thread-51845 - [127.0.0.1]:5703 [dev] [4.2.6-SNAPSHOT] [127.0.0.1]:5703 is SHUTDOWN
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [LifecycleService] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
18:14:07,664  WARN |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Terminating forcefully...
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Shutting down connection manager...
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MockServer] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5702, alive=false}
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MockServer] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5701, alive=false}
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MembershipManager] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 04b9ca9c-a44b-4878-b63e-0cdb505b0b52
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MigrationManager] hz.sharp_hofstadter.migration - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Repartitioning cluster data. Migration tasks count: 6
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [ClusterService] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] 

Members {size:1, ver:5} [
	Member [127.0.0.1]:5701 - bee6fea1-44c0-43fb-ab92-a8f9657c732e this
]

18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [TransactionManagerService] hz.sharp_hofstadter.cached.thread-3 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 04b9ca9c-a44b-4878-b63e-0cdb505b0b52
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MigrationManager] hz.sharp_hofstadter.migration - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Rebalance process was aborted. Ignoring remaining migrations. Will recalculate the new migration plan. (repartitionTime=Thu May 05 18:14:07 UTC 2022, plannedMigrations=6, completedMigrations=0, remainingMigrations=6, totalCompletedMigrations=0, elapsedMigrationOperationTime=0ms, totalElapsedMigrationOperationTime=0ms, elapsedDestinationCommitTime=0ms, totalElapsedDestinationCommitTime=0ms, elapsedMigrationTime=0ms, totalElapsedMigrationTime=0ms)
18:14:07,664  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Shutting down node engine...
18:14:07,665  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [NodeExtension] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Destroying node NodeExtension.
18:14:07,665  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] Hazelcast Shutdown is completed in 1 ms.
18:14:07,665  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [MigrationManager] hz.sharp_hofstadter.migration - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Partition balance is ok, no need to repartition.
18:14:07,665  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [LifecycleService] Thread-51845 - [127.0.0.1]:5702 [dev] [4.2.6-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
18:14:07,665  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [LifecycleService] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
18:14:07,665  WARN |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Terminating forcefully...
18:14:07,665  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Shutting down connection manager...
18:14:07,665  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Shutting down node engine...
18:14:07,666  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [NodeExtension] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Destroying node NodeExtension.
18:14:07,666  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [Node] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] Hazelcast Shutdown is completed in 1 ms.
18:14:07,666  INFO |testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain| - [LifecycleService] Thread-51845 - [127.0.0.1]:5701 [dev] [4.2.6-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain(com.hazelcast.collection.impl.queue.QueuePriorityTest): BuildInfo{version='4.2.6-SNAPSHOT', build='20220505', buildNumber=20220505, revision=e38d1be, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain(com.hazelcast.collection.impl.queue.QueuePriorityTest):'
18:14:05, accumulated pauses: 563 ms, max pause: 404 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueuePriorityTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueuePriorityTest.java
@@ -55,9 +55,13 @@ public class QueuePriorityTest extends HazelcastTestSupport {
     public void before() {
         Config config = smallInstanceConfig();
         config.getQueueConfig("default")
-              .setPriorityComparatorClassName("com.hazelcast.collection.impl.queue.model.PriorityElementComparator");
+                .setPriorityComparatorClassName("com.hazelcast.collection.impl.queue.model.PriorityElementComparator");
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(3);
         HazelcastInstance[] instances = factory.newInstances(config);
+
+        assertClusterSizeEventually(3, instances);
+        warmUpPartitions(instances);
+
         queue = instances[0].getQueue(generateKeyOwnedBy(instances[1]));
         threadPool = Executors.newCachedThreadPool();
     }
@@ -92,7 +96,6 @@ public class QueuePriorityTest extends HazelcastTestSupport {
 
     @Test
     public void testPriorityQueue_whenHighestOfferedFirst_thenTakeHighest() {
-
         PriorityElement elementLow = new PriorityElement(false, 1);
         PriorityElement elementHigh = new PriorityElement(true, 1);
 
@@ -119,7 +122,6 @@ public class QueuePriorityTest extends HazelcastTestSupport {
 
     @Test
     public void testPriorityQueue_whenTwoHighest_thenTakeFirstVersionAgain() {
-
         PriorityElement elementHigh1 = new PriorityElement(true, 1);
         PriorityElement elementHigh2 = new PriorityElement(true, 2);
 

